keras/layers/preprocessing/discretization.py

```diff
@@ -147,7 +147,6 @@ def __init__(
         self.sparse = sparse
 
         if self.bin_boundaries:
-            self.built = True
```

Comment:
Why this change?

Reply:
Explicitly set `self.built=True` in `__init__` will fail the `run_build_asserts` of symbolic call test in `run_layer_test`.
The root cause is that `self.torch_params` will not be initialized.

I'm unsure where the issue is but it should be acceptable to leave `self.built=True` for `build` method.

Reply:
```python
from keras import layers

inputs = layers.Input([2])
layer = layers.Dropout(rate=0.2)
layer(inputs)
assert len(layer.torch_params) == len(layer.variables)

```

This script will fail at the master branch but work fine in this pr.